feat: GlobalExceptionHandler, CommonErrorResponse 구현

gsemily opened the pull request:
## 📌 개요
- global 예외처리 기능 구현
- 공통 에러 응답 클래스 추가

## ✅ 작업 사항
- @ControllerAdvice를 사용한 GlobalExceptionHandler 클래스 추가
- 공통 응답 포맷을 위한 CommonErrorResponse 클래스 생성
- 커스텀 예외 클래스 정의 (BadRequestException, UnauthorizedException, ForbiddenException, NotFoundException)

## 🔍 리뷰 요청 포인트
- 빠진 예외처리 없는지 확인 부탁드립니다.
- response 형식 맞는지 확인 부탁드립니다.

gsemily committed:
feat: GlobalExceptionHandler, CommonErrorResponse 구현

Che0807 commented on src/main/java/com/example/outsourcing_project/global/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,39 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
```
@RestControllerAdvice 또는 @ControllerAdvice 어노테이션 누락되어있는것같습니다.

gsemily replied:
확인 후 반영했습니다

Che0807 commented on src/main/java/com/example/outsourcing_project/global/exception/CommonErrorResponse.java:
```diff
@@ -0,0 +1,19 @@
+package com.example.outsourcing_project.global.exception;
+
+import java.time.LocalDateTime;
+
```
이부분은  Getter가 필요 할것같습니다.

gsemily replied:
확인 후 반영했습니다

Che0807 reviewed (commented):
CommonErrorResponse 이부분은 Getter가 필요 할것같습니다.

Che0807 reviewed (commented):
GlobalExceptionHandler 에@RestControllerAdvice 또는 @ControllerAdvice 어노테이션 누락되어있는것같습니다.

joansdev commented on src/main/java/com/example/outsourcing_project/global/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,39 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
```
@RestControllerAdvice 로 예외 처리 전역화임을 명시해주면 좋을 것 같습니다

joansdev commented on src/main/java/com/example/outsourcing_project/global/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,39 @@
+    public ResponseEntity<CommonErrorResponse> handleCustomExceptions(HttpServletRequest request, RuntimeException ex) {
+        HttpStatus status;
+
+        if (ex instanceof BadRequestException) {
```
반복되는 else-if문보다, 각 예외 클래스와 HTTP 상태 코드를 매핑해 관리하는 게 유지보수나 확장성 측면에서 더 좋을 것 같다는 의견입니다!

예를들면
```
private static final Map<Class<? extends RuntimeException>, HttpStatus> EXCEPTION_STATUS_MAP = Map.of(
        BadRequestException.class, HttpStatus.BAD_REQUEST,
        UnauthorizedException.class, HttpStatus.UNAUTHORIZED,
        ForbiddenException.class, HttpStatus.FORBIDDEN,
        NotFoundException.class, HttpStatus.NOT_FOUND
    );
```
    
```
    public ResponseEntity<CommonErrorResponse> handleCustomExceptions(HttpServletRequest request, RuntimeException ex) {
        HttpStatus status = EXCEPTION_STATUS_MAP.getOrDefault(ex.getClass(), HttpStatus.INTERNAL_SERVER_ERROR);
```
        
-> 예외 추가시 조건문 수정 필요없이 Map에 한 줄 추가로 끝

gsemily replied:
확인 후 반영했습니다.

joansdev reviewed (commented):
ClobalExceptionHandler 코드 리뷰 2가지

gsemily committed:
fix : 어노테이션 추가
- GlobalExceptionHandler에 @RestControllerAdvice 추가
- CommonErrorResponse에 @Getter 추가

gsemily committed:
fix : GlobalExceptionHandler Map 기반으로 리팩토링

Che0807 reviewed (approved):
문제없는것같습니다!
